ls-ui-react-app/src/components/LatticeView.tsx

```diff
@@ -132,57 +133,90 @@ type LatticeViewProps = {
 }
 const LatticeView = ({compilationResult} : LatticeViewProps) => {
     const [selectedSliceNumber, setSelectedSliceNumber] = React.useState<number>(0);
-    const changeSlice = (delta: number) => setSelectedSliceNumber(selectedSliceNumber+delta)
-    const {compilationText, slices} = compilationResult
-    
+    const changeSlice = (delta: number) => {
+        setSelectedSliceNumber(selectedSliceNumber+delta)
+        scrollToLattice()
+    }
+    const {compilation_text, slices} = compilationResult
     const slices_len = slices.length
     // JS object that returns boolean when "previous" or "next" buttons need to be disabled
     const disable = {
         "prev": selectedSliceNumber===0,
         "next": selectedSliceNumber===slices_len-1 
     }
 
-    const latticeRef = useRef<HTMLInputElement>(null);
+    // scroll into Lattice View Section, one time, after compilation is completed
+    const latticeSection = useRef<HTMLInputElement>(null);
     React.useEffect(() => {
-        latticeRef.current && latticeRef!.current!.scrollIntoView();
-    })
+        latticeSection.current && latticeSection!.current!.scrollIntoView();
+    },[])
+
+    // scroll into Lattice View, after slice navigation buttons pressed
+    const lattice = useRef<HTMLInputElement>(null);
+    const scrollToLattice = () => {
+        lattice.current && lattice!.current!.scrollIntoView();
+    }
+
+    // set state of checkbox switch
+    const [checked, setChecked] = useState(false); 
```

Comment:
I think here you want something more specific like `[showCompilationText, setCompilationText]`

Reply:
incorporated these changes